File: js/ResizeSensorPeer.js

fan.afStickySidebar.ResizeSensorPeer = fan.sys.Obj.$extend(fan.sys.Obj);

fan.afStickySidebar.ResizeSensorPeer.prototype.$

fan.afStickySidebar.ResizeSensorPeer.prototype.$ctor = function(self) {
	this.callbackFn = null;
}

fan.afStickySidebar.ResizeSensorPeer.prototype.onResize = function(self, fanFn) {
	this.callbackFn = function() {
		fanFn.call(self.m_elem, self);
	};
	new ResizeSensor(self.m_elem.peer.elem, this.callbackFn);
	return self;
}

fan.afStickySidebar.ResizeSensorPeer.prototype.reset = function(self) {
	ResizeSensor.reset(self.m_elem.peer.elem);
	return self;
}

fan.afStickySidebar.ResizeSensorPeer.prototype.detach = function(self) {
	if (this.callbackFn != null)
		ResizeSensor.detach(self.m_elem.peer.elem, this.callbackFn);
	this.callbackFn = null;
	return self;
}

File: java/ResizeSensorPeer.java
package fan.afStickySidebar;

import fan.sys.*;
import fan.dom.*;

public class ResizeSensorPeer {
	
	private ResizeSensor self;
	
	public ResizeSensorPeer(ResizeSensor self) {
		this.self = self;
	}

	public static ResizeSensorPeer make(ResizeSensor self) throws Exception {
		return new ResizeSensorPeer(self);
	}
	
	public ResizeSensor onResize(Func callback) {
		return self;
	}

	public ResizeSensor reset() {
		return self;
	}

	public ResizeSensor detatch() {
		return self;
	}
}
